app/controllers/items_controller.rb

```diff
@@ -5,23 +5,32 @@ class ItemsController < ApplicationController
   #before_action :current_user.admin?         #WIP must verify that the user is an admin in order for to run any procudure in this controller. An admin
                                               #should be the only user that is able to add/edit/delete items in the database
   
+
   def index                                   #items in library
+    if not params[:filter].nil? and params[:filter]!=""
+      session[:filter] = params[:filter]
+    end
+    if not params[:search].nil?
+      session[:search] = params[:search]
+    end
+    if not session[:search]
+      session[:search]=nil
+    end
     @items = search
-    @page_title = "All library resources"
     
-    if @items.nil?
-      # case when we don't use search, just list all
-      @items = Item.all
-      
-    elsif @items.empty?
+    
+    
+    if @items.empty?
       # case when there is no result
       @page_title = "Must type exact title \"#{params[:search]}\":"
       
+    elsif session[:search].nil? or session[:search]===""
```

Comment:
Not sure if `===` is the operator needed here? That's not very commonly used in ruby.